core/textile/bucket/bucket.go

```diff
@@ -74,15 +115,23 @@ func (b *Bucket) GetData() BucketData {
 	}
 }
 
-func (b *Bucket) getContext(ctx context.Context) (context.Context, *thread.ID, error) {
+func (b *Bucket) GetContext(ctx context.Context) (context.Context, *thread.ID, error) {
 	return b.getBucketContext(ctx, b.root.Name)
 }
 
+func (b *Bucket) GetClient() BucketsClient {
+	return b.bucketsClient
+}
+
 func (b *Bucket) GetThreadID(ctx context.Context) (*thread.ID, error) {
-	_, threadID, err := b.getContext(ctx)
+	_, threadID, err := b.GetContext(ctx)
 	if err != nil {
 		return nil, err
 	}
 
 	return threadID, nil
 }
+
+func (b *Bucket) AttachNotifier(n Notifier) {
+	b.notifier = n
```

Comment:
@dmerrill6 

Should it be possible to attach more than one notifier, eg: one-to-many pins?

Reply:
Don't see a use for that right now, given that if you wanted to do multiple pins you could simply write a custom notifier that does that. We can modify it in the future anyways in case we want to have multiple notifiers.